lib/fluent/plugin/out_mongo.rb

```diff
@@ -213,26 +219,33 @@ def collect_records(chunk)
         if date_keys
           date_keys.each { |date_key|
             begin
-              date_value = record[date_key]
+              date_key_accessor = @accessors[date_key.to_s]
```

Comment:
```
if date_keys
  @accessors.each_pair { |date_key, date_key_accessor|
    date_value = date_key_accessor.call(record)
    # other code
  }
```

Above code is better. `date_value = @accessors[date_key]` is redundant.

Reply:
You are right.. Implemented in latest commit